source/SH3/arc/vfile.cpp

```diff
@@ -100,3 +102,23 @@ std::size_t sh3_arc_vfile::ReadData(void* destination, std::size_t len, read_err
 
     return nbytes;
 }
+
+void sh3_arc_vfile::Dump2Disk()
+{
+    if(!open || buffer.empty())
+    {
+        Log(LogLevel::WARN, "sh3_arc_vfile::Dump2Disk( ): Warning! Attempting to flush unopen or empty buffer to disk!");
+        return;
+    }
+
+    // Firstly, we need to convert the file path to just the filename.
+    std::stringstream ss(fname);
+    std::string item;
+
+    while(getline(ss, item, '/')); // Throw out all strings except the last one (our file name)
```

Comment:
It'd be cheaper to look for the last `/` in the string (see `std::string::rfind`) and passing an iterator pair to `std::ofstream`'s constructor (`begin(fname) + pos` and `end(fname)`).